File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack=new Stack<>();
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='(' || s.charAt(i)=='{' || s.charAt(i)=='['){
                stack.push(s.charAt(i));
            }
            else{
                if(stack.isEmpty()){
                    return false;
                }
                char popped=stack.pop();
                if(popped!=reverseBracket(s.charAt(i))){
                    return false;
                }
            }
        }
        if(!stack.isEmpty()){
            return false;
        }
        return true;
    }
    public char reverseBracket(char c){
        if(c==')'){
            return '(';
        }
        else if(c=='}'){
            return '{';
        }
        else{
            return '[';
        }
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.c
bool isValid(char* s) {
    size_t length=strlen(s);
    char stack[length];
    int size=0;
    for(int i=0;i<length;i++){
        if(s[i]=='(' || s[i]=='{' || s[i]=='['){
            stack[size++]=s[i];
        }
        else{
            if(size==0) return false;
            else{
                if(s[i]==')' && stack[--size]!='('){
                    return false;
                }
                else if(s[i]=='}' && stack[--size]!='{'){
                    return false;
                }
                else if(s[i]==']' && stack[--size]!='['){
                    return false;
                }
            }
        }
    }
    if(size==0){
        return true;
    }
    return false;
}